File: backend/app/schemas/briefings.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.enums import LifecycleStatus


class BriefingScores(BaseModel):
    heat: float | None
    credibility: float | None
    opportunity: float | None
    risk: float | None


class BriefingEvidence(BaseModel):
    raw_item_id: UUID
    title: str
    url: str | None


class BriefingItem(BaseModel):
    intel_file_id: UUID
    title: str
    status: LifecycleStatus
    reason: str
    scores: BriefingScores
    key_evidence: list[BriefingEvidence] = Field(default_factory=list)


class BriefingMeta(BaseModel):
    generated_at: datetime
    window_hours: int
    item_count: int


class BriefingSections(BaseModel):
    new_files: list[BriefingItem] = Field(default_factory=list)
    updated_files: list[BriefingItem] = Field(default_factory=list)
    resurrected: list[BriefingItem] = Field(default_factory=list)
    high_opportunity: list[BriefingItem] = Field(default_factory=list)
    risk_or_noise: list[BriefingItem] = Field(default_factory=list)


class DailyBriefingData(BaseModel):
    meta: BriefingMeta
    overview: str
    sections: BriefingSections
    debug: dict[str, Any] = Field(default_factory=dict)


class WeeklyBriefingMeta(BaseModel):
    generated_at: datetime
    window_days: int
    item_count: int


class WeeklyBriefingSections(BaseModel):
    changed_files: list[BriefingItem] = Field(default_factory=list)
    resurrected: list[BriefingItem] = Field(default_factory=list)
    verified_or_debunked: list[BriefingItem] = Field(default_factory=list)
    opportunity_gainers: list[BriefingItem] = Field(default_factory=list)
    cooling_or_noise: list[BriefingItem] = Field(default_factory=list)


class WeeklyRetrospectiveData(BaseModel):
    meta: WeeklyBriefingMeta
    overview: str
    sections: WeeklyBriefingSections
    debug: dict[str, Any] = Field(default_factory=dict)

File: apps/web/src/types/briefings.ts
export type BriefingScores = {
  heat: number | null;
  credibility: number | null;
  opportunity: number | null;
  risk: number | null;
};

export type BriefingEvidence = {
  raw_item_id: string;
  title: string;
  url: string | null;
};

export type BriefingItem = {
  intel_file_id: string;
  title: string;
  status: string;
  reason: string;
  scores: BriefingScores;
  key_evidence: BriefingEvidence[];
};

export type BriefingSections = {
  new_files: BriefingItem[];
  updated_files: BriefingItem[];
  resurrected: BriefingItem[];
  high_opportunity: BriefingItem[];
  risk_or_noise: BriefingItem[];
};

export type DailyBriefingData = {
  meta: {
    generated_at: string;
    window_hours: number;
    item_count: number;
  };
  overview: string;
  sections: BriefingSections;
  debug: Record<string, unknown>;
};

export type WeeklyBriefingSections = {
  changed_files: BriefingItem[];
  resurrected: BriefingItem[];
  verified_or_debunked: BriefingItem[];
  opportunity_gainers: BriefingItem[];
  cooling_or_noise: BriefingItem[];
};

export type WeeklyRetrospectiveData = {
  meta: {
    generated_at: string;
    window_days: number;
    item_count: number;
  };
  overview: string;
  sections: WeeklyBriefingSections;
  debug: Record<string, unknown>;
};
